client/components/credit-card-form-fields/index.jsx

```diff
@@ -59,54 +80,158 @@ class CreditCardFormFields extends React.Component {
 			[ fieldName ]: maskField( fieldName, previousValue, nextValue ),
 		};
 
-		this.props.onFieldChange( rawDetails, maskedDetails );
+		onFieldChange( rawDetails, maskedDetails );
+	}
+
+	handlePhoneFieldChange = ( { value, countryCode } ) => {
+		this.setState(
+			{
+				phoneCountryCode: countryCode,
+			},
+			() => {
+				this.updateFieldValues( 'phone-number', value );
+			}
+		);
+	};
+
+	handleFieldChange = event => {
+		const { name: fieldName, value: nextValue, options, selectedIndex } = event.target;
+
+		const newState = {};
+
+		if ( fieldName === 'country' ) {
+			newState.countryCode = nextValue;
+			newState.countryName = options[ selectedIndex ].text;
+			if ( ! this.state.phoneCountryCode ) {
+				newState.phoneCountryCode = nextValue;
+			}
+		}
+
+		this.setState( newState, () => this.updateFieldValues( fieldName, nextValue ) );
 	};
 
+	shouldRenderEbanx() {
+		const { countryCode } = this.state;
+		return isEbanx( countryCode );
+	}
+
+	renderEbanxFields() {
+		const { translate, countriesList } = this.props;
+		const { countryCode, phoneCountryCode, countryName } = this.state;
+
+		return [
+			<span key="ebanx-required-fields" className="credit-card-form-fields__info-text">
+				{ translate( 'The following fields are also required for payments in %(countryName)s', {
```

Comment:
:information_source: This string is probably long enough that it doesn't need a **context**. Are you sure you don't want to use a translator comment instead?

Reply:
+1 - I think that both a translator comment and a context are superfluous here - the text is long enough and self-explanatory.